src/decl.cpp

```diff
@@ -798,14 +801,25 @@ void Declarator::InitFromType(const Type *baseType, DeclSpecs *ds) {
             Warning(pos, "\"unmasked\" qualifier is redundant for exported "
                          "functions.");
 
+        if (isUnmangled) {
```

Comment:
A test covering different combinations of attributes/function qualifiers would be good.

Reply:
I have added a couple of tests.